arch/arm/src/max326xx/max32690/max32690_gpio.c

```diff
@@ -0,0 +1,448 @@
+                gpio->padctrl1 &= ~pinmask;
+                gpio->padctrl0 |=  pinmask;
+                gpio->ps       &= ~pinmask;
+                gpio->vssel    &= ~pinmask;
+                break;
+
+            case GPIO_PULLUP_STRONG:
+                gpio->padctrl1 &= ~pinmask;
+                gpio->padctrl0 |=  pinmask;
+                gpio->ps       |=  pinmask;
+                gpio->vssel    &= ~pinmask;
+                break;
+
+            case GPIO_PULL_DOWN_WEAK:
+                gpio->padctrl1 |=  pinmask;
+                gpio->padctrl0 &= ~pinmask;
+                gpio->ps       &= ~pinmask;
+                gpio->vssel    |=  pinmask;
+                break;
+
+            case GPIO_PULL_DOWN_STRONG:
+                gpio->padctrl1 |=  pinmask;
+                gpio->padctrl0 &= ~pinmask;
+                gpio->ps       |=  pinmask;
+                gpio->vssel    |= ~pinmask;
+                break;
+
+            default:
+                DEBUGPANIC();
+                return;
+        }
+}
+
+/****************************************************************************
+ * Name: max32690_gpio_config
+ *
+ * Description:
+ *   Configure a GPIO pin. Config based on bit-encoded descriptions.
+ *
+ ****************************************************************************/
+
+int  max32690_gpio_config(max32690_pinconfig_t pinconf)
+{
+    irqstate_t flags;
+    uint32_t pinmask;
+
+    DEBUGASSERT(pinconf.pin <= GPIO_PINMAX);
+    pinmask = 1 << pinconf.pin;
+
+      /* Modification of all registers must be atomic */
+
+    flags = spin_lock_irqsave(&g_max32690_gpio_lock);
+
+    max32690_gpio_regs_t *gpio;
+    gpio = max32690_get_gpio_bank_regptr(pinconf.gpio_bank);
+
+      /* First, force the pin configuration to the default generic
+       * input state.
+       * So that we know we are starting from a known state.
+       */
+
+    max32690_default(gpio, &pinconf);
+
+      /* Then perform the actual pin configuration.  We need only
+       * to set values that are not in the default, reset state.
+       */
+
+      /* Handle the pin function */
+
+    switch (pinconf.config & GPIO_FUNC_MASK)
+    {
+        case GPIO_IO:
+
+            /* first change to I/O */
+
+            gpio->en0_set = pinmask;
+            gpio->en1_clr = pinmask;
+            gpio->en2_clr = pinmask;
+
+            set_pin_electrical_characteristics(gpio, pinconf.config,
+                                               pinmask);
+
+            if ((pinconf.config & GPIO_OUTPUT_MODE) == GPIO_OUTPUT_MODE)
+            {
+                /* enable pin output mode */
+
+                gpio->outen_set  = pinmask;
+            }
+
+            if ((pinconf.config & GPIO_VALUE_ONE) == GPIO_VALUE_ONE)
+            {
+                gpio->out_set = pinmask;  /* Set output high */
+            }
+
+            break;
+
+      case GPIO_ALT1:
+
+            /* first change to I/O (transition to AF1) */
+
+            gpio->en0_set = pinmask;
+
```

Comment:
remove extra line

Reply:
That line is intentional because modifying en0 carries a specific significance.

Reply:
Ok